### DISCUSSION TITLE
Behaviour like JDBC's "targetServerType"

### DISCUSSION THREAD

**Discussion by cemremengu** [jackc/pgx#2186] [Q&A] [answered]
In JDBC, I can connect to a database and I can provide the targetServerType parameter. The JDBC docs explain this as:

> Allows opening connections to only servers with required state, the allowed values are any, primary, secondary, preferSecondary and preferPrimary. The primary/secondary distinction is currently done by observing if the server allows writes. The value preferSecondary tries to connect to secondary if any are available, otherwise allows falls back to connecting also to primary. The value preferPrimary tries to connect to primary if it is available, otherwise allows falls back to connecting to secondaries available.

It provides a way to configure a HA connection with two hosts, one acting as the primary and one as the secondary. When something is wrong with the primary, the client will automatically switch to the secondary. I know that multiple hosts can be provided in pgx config but there is seems to be no preference for host selection other than reachability. 

Has anyone implemented a behavior similar to this in pgx?

**Comment by jackc** [comment, marked as answer]
I think `Config.ValidateConnect` may be able to do what you want. It is used by pgx to mimic libpq `target_session_attrs` for `read-write`, `read-only`, `primary`, `standby`, and `prefer-standby`.

**Reply by cemremengu** [reply]
Thanks so much, this should work indeed.
